BParkingNano/plugins/TrackMerger.cc

```diff
@@ -53,127 +57,120 @@ class TrackMerger : public edm::global::EDProducer<> {
 
   static void fillDescriptions(edm::ConfigurationDescriptions &descriptions) {}
+  
 
 private:
-  //  const edm::EDGetTokenT<reco::BeamSpot> beamSpotSrc_;
+
   const edm::EDGetTokenT<pat::PackedCandidateCollection> tracksToken_;
   const edm::EDGetTokenT<pat::PackedCandidateCollection> lostTracksToken_;
   const edm::EDGetTokenT<pat::MuonCollection> trgMuonToken_;
 
   //selections                                                                 
-  const double trkPtCut_;
-  const double trkEtaCut_;
-  const double dzTrg_cleaning_;
-  const double drTrg_ProbeCleaning_;
-  const double drTrg_TagCleaning_;
-  const double dcaSig_probe_;
-  const double dcaSig_tag_;
-  const int trkNormChiMin_;
+  const double trkPtCut_;                const double trkEtaCut_;
+  const double dzTrg_cleaning_;          const double drTrg_Cleaning_;
+  // untill the study for this finalized, perhaps we keep it as comment. Then we uncomment it
+ // const double drTrg_ProbeCleaning_;   const double drTrg_TagCleaning_;
+ // const double dcaSig_probe_;         const double dcaSig_tag_;
+  const double dcaSig_;                  const int trkNormChiMin_;
   const int trkNormChiMax_;
 };
 
+
+
+
+
+
 void TrackMerger::produce(edm::StreamID, edm::Event &evt, edm::EventSetup const &stp) const {
-  //get data  
+
+  //input
   edm::ESHandle<MagneticField> bFieldHandle;
   stp.get<IdealMagneticFieldRecord>().get(bFieldHandle);
-  // edm::Handle<reco::BeamSpot> beamSpotHandle;  
-  // evt.getByToken(beamSpotSrc_, beamSpotHandle);
-  // if ( ! beamSpotHandle.isValid() ) {
-  //   edm::LogError("PFCandProducer") << "No beam spot available from EventSetup" ;
-  // }
-  // reco::BeamSpot beamSpot = *beamSpotHandle;
-
   edm::Handle<pat::PackedCandidateCollection> tracks;
   evt.getByToken(tracksToken_, tracks);
-  
   edm::Handle<pat::PackedCandidateCollection> lostTracks;
   evt.getByToken(lostTracksToken_, lostTracks);
-  
-  edm::Handle<pat::MuonCollection> trgMuon;
-  evt.getByToken(trgMuonToken_, trgMuon);
+  edm::Handle<pat::MuonCollection> trgMuons;
+  evt.getByToken(trgMuonToken_, trgMuons);
+
+  //for lost tracks / pf discrimination
+  unsigned int nTracks = tracks->size();
 
-  int nTracks = tracks->size();
-  int nLostTracks = lostTracks->size();      
-  int totalTracks = nTracks + nLostTracks;
+ // do we need all those ints?
+//  int nLostTracks = lostTracks->size();      
+//  int totalTracks = nTracks + nLostTracks;
+ //same as before comment out untill we are sure 
+//  std::unique_ptr<pat::CompositeCandidateCollection> outTag(new pat::CompositeCandidateCollection());
+
+//ok this was CompositeCandidateCollection 
+  std::unique_ptr<pat::CompositeCandidateCollection> tracks_out(new pat::CompositeCandidateCollection);
+
+
+//correct logic but a bit convoluted -> changing to smthn simpler
+ std::vector<pat::PackedCandidate> totalTracks(*tracks);
+
+ totalTracks.insert(totalTracks.end(),lostTracks->begin(),lostTracks->end());
```

Comment:
Add spaces after comas. Also, what is the need of copying them in the new track collection?

Reply:
Is there a way to do it without copying?